redis/client.py

```diff
@@ -2226,6 +2226,10 @@ def _execute(self, connection, command, *args):
     def parse_response(self, block=True, timeout=0):
         "Parse the response from a publish/subscribe command"
         connection = self.connection
+        if connection is None:
+            raise RuntimeError(
```

Comment:
* Is this a Programming Error? Is it at all supported to call `get_message` without first subscribing?
* If so, do we want an assertion here? Do we expect people ever to `try...except` on this, or should they just fix it?
* Could we just `try` the `connection.can_read` and catch the `AttributeError`, rather than checking first?